Make all the things

martinpitt opened the pull request:
See individual commits for details. I tested all of make-{dist,srpm,rpm,deb} from a completely clean git checkout, with downloading and with self-building.

martinpitt commented:
@marusak: This should also fix your make-rpms woes.

KKoukiou commented on tools/make-debs:
```diff
@@ -0,0 +1,78 @@
+# plug in our version
+sed -i "1 s/(0-/(${version}-/" debian/changelog
+dpkg-buildpackage -us -uc -b -j$(nproc)
+cd "$olddir"
```
`cd -` also works here, just to avoid storing olddir.

martinpitt replied:
Done, thanks!

martinpitt commented:
I fixed up the messaging harder, and added another fix for a bug that @marusak just found - we shouldn't download a tarball if there are local uncommitted changes.

marusak commented:
Nice, thank you for fixing up both issues I've struggled with today morning!

KKoukiou reviewed (approved):
I have already tests the make-debs in my livetest branch btw, works as expected.
